### PR TITLE
erlang 22.0

### PR DESCRIPTION
**Pull request by tomciopp**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by tomciopp** [comment]
Running `brew test erlang` returns the following error: 

    Error: Testing requires the latest version of erlang

Running `brew audit --strict erlang` returns the following error:

    erlang:
    * Formula has invalid versioned aliases:
      erlang@22
    Error: 1 problem in 1 formula detected

File: Aliases/erlang@21 [hunk 1]
```diff
@@ -1 +1 @@
-../Formula/erlang.rb
\ No newline at end of file
+/Users/thomascioppettini/oss/homebrew-core/Formula/erlang@21.rb
```
Comment (fxcoudert) [inline Aliases/erlang@21]: This alias should be removed.

Comment (tomciopp) [reply Aliases/erlang@21]: haha oops will fix

**Review by fxcoudert — CHANGES_REQUESTED** [review]
We have too many versioned erlang formulas in core, we need to remove some of the older ones (17 and 18).

File: Aliases/erlang@21 (hunk 1 above)
Comment (tomciopp) [reply Aliases/erlang@21]: should it be removed entirely or fixed?

Comment (fxcoudert) [reply Aliases/erlang@21]: Removed. In `Aliases` we only have one alias, to the current version formula.

**Comment by tomciopp** [comment]
@fxcoudert Updated code

File: Formula/erlang@21.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
+class ErlangAT21 < Formula
+  desc "Programming language for highly scalable real-time systems"
+  homepage "https://www.erlang.org/"
+  # Download tarball from GitHub; it is served faster than the official tarball.
+  url "https://github.com/erlang/otp/archive/OTP-21.3.8.tar.gz"
+  sha256 "e20df59eac5ec0f3d47cb775eb7cfb20438df24d93ba859959a18fe07abf3e6e"
+  head "https://github.com/erlang/otp.git"
+
+  bottle do
```
Comment (fxcoudert) [inline Formula/erlang@21.rb]: Remove the bottle block

File: Formula/erlang@21.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
+class ErlangAT21 < Formula
+  desc "Programming language for highly scalable real-time systems"
+  homepage "https://www.erlang.org/"
+  # Download tarball from GitHub; it is served faster than the official tarball.
+  url "https://github.com/erlang/otp/archive/OTP-21.3.8.tar.gz"
+  sha256 "e20df59eac5ec0f3d47cb775eb7cfb20438df24d93ba859959a18fe07abf3e6e"
+  head "https://github.com/erlang/otp.git"
```
Comment (fxcoudert) [inline Formula/erlang@21.rb]: Remove head

File: Formula/erlang@21.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+  bottle do
+    cellar :any
+    sha256 "5cbda1e3a4c354e954d006991bdae1bf740a73c8f5df36dda740c0af2c4a47b9" => :mojave
+    sha256 "608dae8316ef3ddbd34935c723f229ee85715547eb0d3623bbb67458ebd74f69" => :high_sierra
+    sha256 "9acb1df98cb7e3d88fc01b20123e1031e4cd4c0a9c051dc81fb889c3a05018ba" => :sierra
+  end
+
```
Comment (fxcoudert) [inline Formula/erlang@21.rb]: Add `keg_only :versioned_formula` like other versioned erlang formulas

**Review by fxcoudert — CHANGES_REQUESTED** [review]
We need to determine whether `riak` can be compiled with a more recent erlang version

**Comment by tomciopp** [comment]
@fxcoudert https://github.com/basho/riak/issues/978

**Comment by tomciopp** [comment]
@fxcoudert So it looks like we still need to keep the older versions of erlang based on the response from the riak people. If I add 17 back, should I add 18 as well?

**Comment by tomciopp** [comment]
@fxcoudert Bumping this, how should we proceed?

**Comment by fxcoudert** [comment]
@tomciopp restore erlang 17, leave erlang 18 out, and we should be good

**Comment by tomciopp** [comment]
@fxcoudert Done

**Comment by archseer** [comment]
@tomciopp Can you bump this to OTP 22.0.1? It was released a day or two after to fix an important ssh issue: https://github.com/erlang/otp/releases/tag/OTP-22.0.1

erlang.org is just always a bit behind on minor release announcements

**Comment by tomciopp** [comment]
@archseer Will update.

**Comment by plus3x** [comment]
@tomciopp Correct end of files please(https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/42314/version=high_sierra/testReport/)

```
erlang@17:
  * File should end with a newline
```

**Comment by fxcoudert** [comment]
Please squash your changes into three commits:
- `erlang@18: delete`
- `erlang 22.0.1` for the main formula update (and alias rename)
- `erlang@21 21.3.8 (new formula)` for the new Erlang 21 formula

**Comment by tomciopp** [comment]
@fxcoudert Updated

**Comment by tomciopp** [comment]
@fxcoudert bumping, can we merge?

**Review by zbeekman — APPROVED** [review]
LGTM

**Comment by zbeekman** [comment]
Thanks for your contribution to Homebrew! 🎉 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**